daemon/c_vol.c

```diff
@@ -1352,26 +1469,10 @@ c_vol_start_child(c_vol_t *vol)
 		goto error;
 	}
 
-	if (chdir(vol->root) < 0) {
-		ERROR_ERRNO("Could not chdir to root dir %s for container start", vol->root);
-		goto error;
-	}
-
-	// mount namespace handles chroot jail breaks
-	if (mount(".", "/", NULL, MS_MOVE, NULL) < 0) {
-		ERROR_ERRNO("Could not move mount for container start");
-		goto error;
-	}
-
-	if (chroot(".") < 0) {
-		ERROR_ERRNO("Could not chroot to . for container start");
-		goto error;
-	}
-
-	if (chdir("/") < 0) {
-		ERROR_ERRNO("Could not chdir to / for container start");
-		goto error;
-	}
+	if (cmld_is_hostedmode_active() && !container_has_userns(vol->container))
```

Comment:
should this be || ?

Reply:
nop, pivot root would only work for non userns in hosted mode.